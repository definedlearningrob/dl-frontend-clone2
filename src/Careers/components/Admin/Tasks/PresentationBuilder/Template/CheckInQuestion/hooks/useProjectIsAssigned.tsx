import { useMemo } from 'react';
import { useParams } from 'react-router-dom';
import { useQuery } from '@apollo/client';
import { isEmpty } from 'lodash-es';

import teamProjectQuery from '@pbl/graphql/student/queries/teamProject';
import projectQuery from '@pbl/graphql/student/queries/project';

import { useRole } from '@shared/hooks/useRole';

export const useProjectIsAssigned = () => {
  const { isStudent } = useRole();
  const { projectId, teamId } = useParams<{ projectId: string; teamId: string }>();

  const properQuery = useMemo(() => (teamId ? teamProjectQuery : projectQuery), [teamId]);

  const queryVariables = useMemo(
    () =>
      teamId
        ? { id: projectId, teamId, track: true, trackPresentation: true }
        : { id: projectId, track: true, trackPresentation: true },
    [teamId]
  );

  const { data, error, loading } = useQuery(properQuery, {
    variables: queryVariables,
    skip: !isStudent,
  });

  if (!isStudent || loading || error) {
    return false;
  }

  return !isEmpty(data?.project?.assignedAt);
};
