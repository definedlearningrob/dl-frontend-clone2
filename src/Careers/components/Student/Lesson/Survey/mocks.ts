import lessonInCourseQuery from '@dc/graphql/student/queries/lessonInCourse';
import { CREATE_CAREER_REVIEW_SURVEY_ATTEMPT } from '@dc/graphql/student/mutations/createCareerReviewSurveyAttempt';
import createCareerReviewSurveyAnswersMutation from '@dc/graphql/student/mutations/createCareerReviewSurveyAnswers';
import currentCoursesQuery from '@dc/graphql/student/queries/currentCourses';
import { SURVEY_QUESTION_TYPE } from '@dc/components/Student/Lesson/Survey/types';

export const createAttemptSpy = jest.fn();
export const lessonInCourseSpy = jest.fn();

export let createCareerReviewSurveyAnswersSpy = jest.fn();
export let currentCoursesSpy = jest.fn();

export const mocks = [
  {
    request: {
      query: lessonInCourseQuery,
      variables: { courseId: '2', lessonId: '111', track: true },
    },
    result: () => {
      lessonInCourseSpy();

      return {
        data: {
          course: {
            careerName: 'Dental Laboratory Technicians',
            content: {
              checkIns: [],
              items: [],
            },
            hasInstitutionsInStudentState: true,
            id: '2',
            lesson: {
              assignments: [],
              attachments: [],
              careerReviewSurvey: {
                version: 1,
                performed: false,
                questions: [
                  {
                    answer: [],
                    id: '1',
                    options: [
                      {
                        option: 'Yes, definitely',
                        step: 1,
                      },
                      {
                        option: 'Definitely not',
                        step: 2,
                      },
                    ],
                    question: 'Can blind people see their dreams?',
                    type: 'single_choice' as SURVEY_QUESTION_TYPE,
                  },
                  {
                    answer: [],
                    id: '2',
                    options: [
                      {
                        option: 'Very Interested',
                        step: 1,
                      },
                      {
                        option: 'Not at all Interested',
                        step: 2,
                      },
                      {
                        option: 'Other',
                        step: 3,
                      },
                    ],
                    question: 'How interested are you in the Pathway this course is a part of?',
                    type: 'multiple_choice' as SURVEY_QUESTION_TYPE,
                  },
                ],
                __typename: 'CareerReviewSurvey',
              },
              checkInGroups: [],
              checkInQuestions: [],
              description: {
                introduction: null,
                goal: null,
                role: null,
                audience: null,
                situation: null,
                __typename: 'LessonDescription',
              },
              hasPresentation: false,
              id: '609',
              imageUrl: 'http://test.com/image.png',
              name: 'Career Review Survey',
              externalPresentations: [],
              researchLinks: [],
              texts: [],
              type: 'career_review_survey',
              videos: [],
              vocabularies: [],
              __typename: 'Lesson',
            },
            lessons: [
              {
                id: '782',
                step: 4,
                careerReviewSurvey: null,
              },
              {
                id: '609',
                step: 5,
                careerReviewSurvey: {
                  performed: false,
                  version: 1,
                  __typename: 'CareerReviewSurvey',
                },
              },
              {
                id: '275',
                step: 1,
                careerReviewSurvey: null,
              },
              {
                id: '331',
                step: 2,
                careerReviewSurvey: null,
              },
              {
                id: '668',
                step: 3,
                careerReviewSurvey: null,
              },
            ],
            name: 'Dental Laboratory Technician',
            progress: {
              submitted: 10,
              total: 10,
            },
            reviewSurvey: {
              questions: [
                {
                  id: '1',
                },
                {
                  id: '2',
                },
                {
                  id: '3',
                },
                {
                  id: '4',
                },
                {
                  id: '5',
                },
                {
                  id: '6',
                },
                {
                  id: '7',
                },
                {
                  id: '8',
                },
                {
                  id: '9',
                },
                {
                  id: '10',
                },
                {
                  id: '11',
                },
              ],
            },
          },
        },
      };
    },
  },
  {
    request: {
      query: CREATE_CAREER_REVIEW_SURVEY_ATTEMPT,
      variables: { input: { contextType: 'COURSE', contextId: '1' } },
    },
    result: () => {
      createAttemptSpy();

      return {
        data: {
          createCareerReviewSurveyAttempt: {
            careerReviewSurveyAttempt: {
              id: '2',
              status: 'IN_PROGRESS',
            },
          },
        },
      };
    },
  },
  {
    request: {
      query: createCareerReviewSurveyAnswersMutation,
      variables: {
        input: {
          contextId: '1',
          contextType: 'COURSE',
          answers: [
            { questionId: '1', answer: ['Yes, definitely'] },
            { questionId: '2', answer: ['Very Interested'] },
          ],
        },
      },
    },
    result: () => {
      createCareerReviewSurveyAnswersSpy();

      return {
        data: {
          createCareerReviewSurveyAnswers: {
            status: 'success',
          },
        },
      };
    },
  },
  {
    request: {
      query: currentCoursesQuery,
    },
    result: () => {
      currentCoursesSpy();

      return {
        data: {
          currentCourses: {
            id: '1',
            imageUrl: '',
            name: '',
            progress: {
              submitted: 2,
              total: 5,
            },
            status: 'success',
            pathway: {
              name: '',
            },
            thumbnailUrl: '',
            type: 'survey',
          },
        },
      };
    },
  },
];
