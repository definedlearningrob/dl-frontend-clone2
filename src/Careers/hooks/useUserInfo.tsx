import {
  createContext,
  Dispatch,
  ReactNode,
  SetStateAction,
  useContext,
  useEffect,
  useState,
} from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useApolloClient, useLazyQuery } from '@apollo/client';

import studentInfoQuery, {
  TStudentInfo,
  TStudentInfoData,
} from '@dc/graphql/student/queries/userInfo';
import userInfoQuery, { TUserInfo, TUserInfoData } from '@dc/graphql/user/queries/userInfo';
import { impersonateAction, logoutAction } from '@dc/redux/session/actions';
import { ROLES } from '@dc/resources/constants';
import { SessionState } from '@dc/redux/session/reducer';

type TContextState = THookState & {
  setState: Dispatch<SetStateAction<THookState>>;
  pollInterval: number;
  setPollInterval: (interval: number) => void;
};

type THookState = {
  userInfo: TUserInfo | TStudentInfo | null;
  loading: boolean;
};
const UserInfoContext = createContext({} as TContextState);

type Props = { value?: { userInfo: TUserInfo | TStudentInfo }; children: ReactNode };

// eslint-disable-next-line no-undef
const defaultPollInterval = import.meta.env.VITE_POLLING_INTERVAL as any as number;

export function UserInfoProvider({ value, children }: Props) {
  const [pollInterval, setPollInterval] = useState(defaultPollInterval);
  const [{ userInfo, loading }, setState] = useState<THookState>({
    userInfo: value ? value.userInfo : null,
    loading: true,
  });

  return (
    <UserInfoContext.Provider
      value={{ userInfo, loading, pollInterval, setPollInterval, setState, ...value }}>
      {children}
    </UserInfoContext.Provider>
  );
}

function useUserInfo<T extends TUserInfo | TStudentInfo>() {
  const { loading, setState, userInfo, pollInterval, setPollInterval } =
    useContext(UserInfoContext);

  const { user } = useSelector(({ session }: { session: SessionState }) => session);
  const query = user?.type === 'student' ? studentInfoQuery : userInfoQuery;
  const dispatch = useDispatch();
  const client = useApolloClient();

  const [queryUser] = useLazyQuery<TUserInfoData | TStudentInfoData>(query, {
    // eslint-disable-next-line no-undef
    pollInterval,
    notifyOnNetworkStatusChange: true,
  });

  useEffect(() => {
    if (user && !userInfo) {
      // this variables are wrong, only to force refetch of userinfo
      queryUser({ variables: { username: user.username } }).then((response) => {
        if (response.data) {
          setState({ userInfo: response.data.userInfo, loading: false });
        }
      });
    }
  }, [user]);

  const [refreshUser] = useLazyQuery<TUserInfoData | TStudentInfoData>(query, {
    fetchPolicy: 'network-only',
    onCompleted: (data) => setState({ userInfo: data.userInfo, loading: false }),
  });

  const logout = async () => {
    setState({ userInfo: null, loading: false });
    dispatch(logoutAction());

    await client.clearStore();
  };

  const impersonateStart = async (userType: 'User' | 'Student', studentUuid: string) => {
    setState({ userInfo: null, loading: true });
    await dispatch(impersonateAction('start', userType, studentUuid));
    await client.clearStore();
  };

  const impersonateStop = async () => {
    setState({ userInfo: null, loading: true });
    await dispatch(impersonateAction('stop'));
    await client.clearStore();
  };

  const stopPolling = () => setPollInterval(86400000);
  const startPolling = (interval = defaultPollInterval) => setPollInterval(interval);

  const isSelf = (uuid: string) => uuid === userInfo?.uuid;

  return {
    loading,
    isSelf,
    refreshUser,
    isSystemAdminUser: getRole(userInfo) === ROLES.SYSTEM_ADMIN,
    userInfo: userInfo as T,
    logout,
    impersonateStart,
    startPolling,
    impersonateStop,
    stopPolling,
  };
}

export const getRole = (userInfo: TUserInfo | TStudentInfo | null) => {
  if (!userInfo) return undefined;

  if ('role' in userInfo) {
    return userInfo.role;
  }

  return undefined;
};

export default useUserInfo;
