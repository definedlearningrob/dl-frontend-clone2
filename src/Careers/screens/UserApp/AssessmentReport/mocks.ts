import { MockedResponse } from '@apollo/client/testing';

import {
  ASSESSMENT_REPORT,
  TAssessmentReportData,
} from '@dc/graphql/user/queries/reportsAssessmentReport';
import {
  ASSESSMENT_REPORT_FILTERS,
  TAssessmentReportFiltersData,
} from '@dc/graphql/user/queries/assessmentReportFilters';

export const assessmentReportMock: MockedResponse<TAssessmentReportData> = {
  request: {
    query: ASSESSMENT_REPORT,
    variables: {
      filter: { schoolYear: 2023 },
    },
  },
  result: {
    data: {
      reports: {
        assessmentReport: {
          studentsCount: 300,
          summary: {
            assessmentCompleted: 100,
            assessmentTaken: 200,
          },
          highSchoolInterests: [
            { category: 'Artistic', score: 1 },
            { category: 'Conventional', score: 4 },
            { category: 'Enterprising', score: 4 },
            { category: 'Investigative', score: 6 },
            { category: 'Realistic', score: 6 },
            { category: 'Social', score: 9 },
          ],
          middleSchoolInterests: [
            { category: 'Creator', score: 2 },
            { category: 'Doer', score: 3 },
            { category: 'Helper', score: 4 },
            { category: 'Organizer', score: 4 },
            { category: 'Persuader', score: 6 },
            { category: 'Thinker', score: 8 },
          ],
          middleSchoolWorkValues: [
            { category: 'Achievement', averageTokens: 10 },
            { category: 'Independence', averageTokens: 14 },
            { category: 'Recognition', averageTokens: 4 },
            { category: 'Relationships', averageTokens: 6 },
            { category: 'Support', averageTokens: 28 },
            { category: 'Working Conditions', averageTokens: 15 },
          ],
          highSchoolWorkValues: [
            { category: 'Achievement', averageTokens: 20 },
            { category: 'Independence', averageTokens: 10 },
            { category: 'Recognition', averageTokens: 7 },
            { category: 'Relationships', averageTokens: 30 },
            { category: 'Support', averageTokens: 6 },
            { category: 'Working Conditions', averageTokens: 1 },
          ],
          highSchoolStudyPreferences: [
            {
              area: 'Math',
              results: {
                position1: 6,
                position2: 0,
                position3: 0,
                position4: 2,
                position5: 0,
                position6: 0,
                position7: 0,
              },
            },
            {
              area: 'Science',
              results: {
                position1: 0,
                position2: 0,
                position3: 0,
                position4: 0,
                position5: 0,
                position6: 0,
                position7: 8,
              },
            },
            {
              area: 'Social Studies',
              results: {
                position1: 2,
                position2: 5,
                position3: 0,
                position4: 1,
                position5: 0,
                position6: 0,
                position7: 0,
              },
            },
            {
              area: 'Language Arts',
              results: {
                position1: 0,
                position2: 0,
                position3: 2,
                position4: 0,
                position5: 0,
                position6: 6,
                position7: 0,
              },
            },
            {
              area: 'Computers',
              results: {
                position1: 0,
                position2: 1,
                position3: 5,
                position4: 0,
                position5: 1,
                position6: 1,
                position7: 0,
              },
            },
            {
              area: 'Creative Arts',
              results: {
                position1: 0,
                position2: 1,
                position3: 0,
                position4: 0,
                position5: 6,
                position6: 1,
                position7: 0,
              },
            },
            {
              area: 'Trades',
              results: {
                position1: 0,
                position2: 1,
                position3: 1,
                position4: 5,
                position5: 1,
                position6: 0,
                position7: 0,
              },
            },
          ],
          middleSchoolStudyPreferences: [
            {
              area: 'Language Arts',
              results: {
                position1: 0,
                position2: 1,
                position3: 0,
                position4: 1,
                position5: 0,
              },
            },
            {
              area: 'Math',
              results: {
                position1: 0,
                position2: 0,
                position3: 2,
                position4: 0,
                position5: 0,
              },
            },
            {
              area: 'Science',
              results: {
                position1: 0,
                position2: 0,
                position3: 0,
                position4: 0,
                position5: 2,
              },
            },
            {
              area: 'Social Studies',
              results: {
                position1: 2,
                position2: 0,
                position3: 0,
                position4: 0,
                position5: 0,
              },
            },
            {
              area: 'Creative Arts',
              results: {
                position1: 0,
                position2: 1,
                position3: 0,
                position4: 1,
                position5: 0,
              },
            },
          ],
          clusterRecommendationCounts: [
            {
              cluster: {
                name: 'Education & Training',
                id: '3',
                pathways: [
                  {
                    name: 'Administration and Administrative Support',
                    id: '14',
                  },
                  {
                    name: 'Professional Support Services',
                    id: '15',
                  },
                  {
                    name: 'Teaching/Training',
                    id: '16',
                  },
                ],
              },
              pathwayRecommendationCounts: [
                {
                  recommendationsCount: 2,
                  pathway: {
                    name: 'Teaching/Training',
                    id: '16',
                  },
                },
                {
                  recommendationsCount: 1,
                  pathway: {
                    name: 'Administration and Administrative Support',
                    id: '14',
                  },
                },
                {
                  recommendationsCount: 1,
                  pathway: {
                    name: 'Professional Support Services',
                    id: '15',
                  },
                },
              ],
              recommendationsCount: 4,
            },
            {
              cluster: {
                name: 'Business Management & Administration',
                id: '13',
                pathways: [
                  {
                    name: 'General Management',
                    id: '61',
                  },
                  {
                    name: 'Business Information Management',
                    id: '62',
                  },
                  {
                    name: 'Human Resources Management',
                    id: '63',
                  },
                  {
                    name: 'Operations Management',
                    id: '64',
                  },
                  {
                    name: 'Administrative Support',
                    id: '65',
                  },
                ],
              },
              pathwayRecommendationCounts: [
                {
                  recommendationsCount: 3,
                  pathway: {
                    name: 'Administrative Support',
                    id: '65',
                  },
                },
                {
                  recommendationsCount: 1,
                  pathway: {
                    name: 'Business Information Management',
                    id: '62',
                  },
                },
              ],
              recommendationsCount: 4,
            },
            {
              cluster: {
                name: 'Manufacturing',
                id: '9',
                pathways: [
                  {
                    name: 'Production',
                    id: '41',
                  },
                  {
                    name: 'Manufacturing Production Process Development',
                    id: '42',
                  },
                  {
                    name: 'Maintenance, Installation & Repair',
                    id: '43',
                  },
                  {
                    name: 'Quality Assurance',
                    id: '44',
                  },
                  {
                    name: 'Logistics & Inventory Control',
                    id: '45',
                  },
                  {
                    name: 'Health, Safety & Environmental Assurance',
                    id: '46',
                  },
                ],
              },
              pathwayRecommendationCounts: [
                {
                  recommendationsCount: 2,
                  pathway: {
                    name: 'Production',
                    id: '41',
                  },
                },
                {
                  recommendationsCount: 2,
                  pathway: {
                    name: 'Manufacturing Production Process Development',
                    id: '42',
                  },
                },
              ],
              recommendationsCount: 4,
            },
          ],
        },
      },
    },
  },
};

export const assessmentReportFiltersMock: MockedResponse<TAssessmentReportFiltersData> = {
  request: {
    query: ASSESSMENT_REPORT_FILTERS,
    variables: {
      filters: {},
      entityFilter: { nameCont: '' },
      schoolClassFilter: { nameCont: '' },
      userFilter: { fullNameCont: '' },
    },
  },
  result: {
    data: {
      assessmentReportFilters: {
        entities: {
          nodesCount: 1,
          pagesCount: 1,
          nodes: [{ uuid: '1234-5678', name: 'City High School' }],
        },
        gradeLevels: ['10', '11', '12', 'Postsecondary'],
        users: {
          nodesCount: 1,
          pagesCount: 1,
          nodes: [{ uuid: '9876-5432', fullName: 'Anna Cityteach' }],
        },
        schoolClasses: {
          nodesCount: 0,
          pagesCount: 0,
          nodes: [],
        },
      },
    },
  },
};
