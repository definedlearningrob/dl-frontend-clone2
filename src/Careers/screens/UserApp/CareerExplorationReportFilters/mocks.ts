import { MockedResponse } from '@apollo/client/testing';
import {
  CareerExplorationReportFiltersDocument,
  CareerExplorationReportSummaryDocument,
  CareerExplorationReportVisitCountsDocument,
} from '@graphql/dc/users/hooks';

import { TCareerExplorationReportFiltersData } from '@dc/graphql/user/queries/careerExplorationReportFilters';

export const careerExplorationReportSummaryMock = {
  request: {
    query: CareerExplorationReportSummaryDocument,
    variables: {
      filter: { schoolYear: 2023, schoolClassUuids: undefined },
    },
  },
  result: {
    data: {
      reports: {
        pathwayReport: {
          summary: {
            engagementsCount: 10,
            clusterSCount: 2,
            pathwaysCount: 5,
            studentCount: 12,
            usersCount: 3,
          },
        },
      },
    },
  },
};

export const visitCountsSpy = jest.fn();

export const careerExplorationReportVisitCountsMock = {
  request: {
    query: CareerExplorationReportVisitCountsDocument,
    variables: {
      filter: { schoolYear: 2023, schoolClassUuids: undefined },
    },
  },
  result() {
    visitCountsSpy();

    return {
      data: {
        reports: {
          pathwayReport: [
            {
              visitsCount: 396,
              cluster: {
                name: 'Agriculture, Food & Natural Resources',
                id: '1',
                __typename: 'Cluster',
              },
              pathwayVisitCounts: [
                {
                  visitsCount: 54,
                  pathway: {
                    cluster: {
                      name: 'Agriculture, Food & Natural Resources',
                      id: '1',
                    },
                    id: '5',
                    name: 'Natural Resources Systems',
                  },
                },
                {
                  visitsCount: 55,
                  pathway: {
                    cluster: {
                      name: 'Agriculture, Food & Natural Resources',
                      id: '1',
                    },
                    id: '2',
                    name: 'Animal Systems',
                  },
                },
                {
                  visitsCount: 54,
                  pathway: {
                    cluster: {
                      name: 'Agriculture, Food & Natural Resources',
                      id: '1',
                    },
                    id: '6',
                    name: 'Plant Systems',
                  },
                },
              ],
            },
            {
              visitsCount: 10,
              cluster: {
                name: 'Transportation, Distribution & Logistics',
                id: '11',
              },
              pathwayVisitCounts: [
                {
                  visitsCount: 2,
                  pathway: {
                    cluster: {
                      name: 'Transportation, Distribution & Logistics',
                      id: '11',
                    },
                    id: '52',
                    name: 'Facility and Mobile Equipment Maintenance',
                  },
                },
                {
                  visitsCount: 3,
                  pathway: {
                    cluster: {
                      name: 'Transportation, Distribution & Logistics',
                      id: '11',
                    },
                    id: '49',
                    name: 'Transportation Operations',
                  },
                },
                {
                  visitsCount: 5,
                  pathway: {
                    cluster: {
                      name: 'Transportation, Distribution & Logistics',
                      id: '11',
                    },
                    id: '51',
                    name: 'Warehousing & Distribution Center Operations',
                  },
                },
              ],
            },
          ],
        },
      },
    };
  },
};

export const careerExplorationReportFiltersMock: MockedResponse<TCareerExplorationReportFiltersData> =
  {
    request: {
      query: CareerExplorationReportFiltersDocument,
      variables: {
        filters: {},
        entityFilter: { nameCont: '' },
        schoolClassFilter: { nameCont: '' },
        userFilter: { fullNameCont: '' },
      },
    },
    result: {
      data: {
        pathwayReportFilters: {
          entities: {
            nodesCount: 1,
            pagesCount: 1,
            nodes: [{ uuid: '1234-5678', name: 'City High School' }],
          },
          gradeLevels: ['10', '11', '12', 'Postsecondary'],
          users: {
            nodesCount: 1,
            pagesCount: 1,
            nodes: [{ uuid: '9876-5432', fullName: 'Anna Cityteach' }],
          },
          schoolClasses: {
            nodesCount: 0,
            pagesCount: 0,
            nodes: [],
          },
        },
      },
    },
  };
