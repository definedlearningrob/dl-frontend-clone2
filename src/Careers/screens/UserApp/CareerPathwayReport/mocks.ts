import { MockedResponse } from '@apollo/client/testing';
import { CareerExplorationReportFullDataDocument } from '@graphql/dc/users/hooks';
import { CareerExplorationReportFullDataQuery } from '@graphql/dc/users/operations';

export const careerExplorationReportMock: MockedResponse<CareerExplorationReportFullDataQuery> = {
  request: {
    query: CareerExplorationReportFullDataDocument,
    variables: {
      filter: { schoolYear: 2023 },
    },
  },
  result: {
    data: {
      reports: {
        pathwayReport: {
          visitResults: {
            nodesCount: 1,
            pagesCount: 1,
            nodes: [
              {
                clusterNames: ['Education & Training'],
                isEnrolled: true,
                pathwayNames: ['Teaching/Training'],
                resourceId: '16',
                resourceName: 'Teaching/Training',
                resourceType: 'Pathway',
                visitorId: '1234',
                visitorName: 'Jeremiah Bell',
                visitorEmail: 'jeremiah.bell@example.com',
                visitorSisId: null,
                visitorType: 'Student',
                visitsCount: 5,
              },
            ],
          },
        },
      },
    },
  },
};
