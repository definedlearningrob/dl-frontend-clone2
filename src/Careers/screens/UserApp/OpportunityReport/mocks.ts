import { MockedResponse } from '@apollo/client/testing';
import { OpportunityReportFullDataDocument } from '@graphql/dc/users/hooks';
import { OpportunityReportFullDataQuery } from '@graphql/dc/users/operations';
import { ApplicationStatus, OpportunityTypes } from '@graphql/dc/shared/types';
import { ClustersQuery, ClustersQueryVariables } from '@graphql/dc/shared/operations';
import { ClustersDocument } from '@graphql/dc/shared/hooks';

export const clustersMock: MockedResponse<ClustersQuery, ClustersQueryVariables> = {
  request: {
    query: ClustersDocument,
    variables: {},
  },
  result: {
    data: {
      clusters: [
        {
          id: '1',
          name: 'Agriculture, Food & Natural Resources',
          pathways: [
            {
              id: '1',
              name: 'Agribusiness Systems',
            },
            {
              id: '2',
              name: 'Animal Systems',
            },
          ],
        },
      ],
    },
  },
};

export const opportunityResultsMock: MockedResponse<OpportunityReportFullDataQuery> = {
  request: {
    query: OpportunityReportFullDataDocument,
    variables: {
      filter: { schoolYear: 2023, schoolClassUuids: undefined },
      resultsFilter: {},
      page: 1,
      perPage: 25,
    },
  },
  result: {
    data: {
      reports: {
        opportunityReport: {
          results: {
            nodesCount: 3,
            pagesCount: 1,
            nodes: [
              {
                applicationDeadline: null,
                applicationStatus: ApplicationStatus.ACCEPTED,
                assignmentsSubmitted: 3,
                assignmentsToSubmit: 6,
                checkInsSubmitted: 0,
                checkInsToSubmit: 0,
                clusterNames: ['Marketing', 'Transportation, Distribution & Logistics'],
                isFavorite: false,
                opportunityName: 'Test',
                opportunityPartnerNames: [],
                opportunityType: OpportunityTypes.VIRTUAL_INTERNSHIP,
                pathwayNames: ['Travel & Tourism', 'Warehousing & Distribution Center Operations'],
                studentGradeLevel: '3',
                studentId: '10458251',
                studentName: 'Amanda Brown',
                studentSisId: '242424',
              },
              {
                applicationDeadline: null,
                applicationStatus: ApplicationStatus.ACCEPTED,
                assignmentsSubmitted: 0,
                assignmentsToSubmit: 0,
                checkInsSubmitted: 0,
                checkInsToSubmit: 0,
                clusterNames: [
                  'Agriculture, Food & Natural Resources',
                  'Architecture & Construction',
                ],
                isFavorite: false,
                opportunityName: 'Test 2',
                opportunityPartnerNames: [],
                opportunityType: OpportunityTypes.VIRTUAL_INTERNSHIP,
                pathwayNames: [
                  'A/V Technology & Film',
                  'Administration and Administrative Support',
                ],
                studentGradeLevel: '7',
                studentId: '10458576',
                studentName: 'Jon James',
                studentSisId: null,
              },
              {
                applicationDeadline: '2055-01-01',
                applicationStatus: ApplicationStatus.ACCEPTED,
                assignmentsSubmitted: null,
                assignmentsToSubmit: null,
                checkInsSubmitted: null,
                checkInsToSubmit: null,
                clusterNames: [
                  'Agriculture, Food & Natural Resources',
                  'Business Management & Administration',
                ],
                isFavorite: true,
                opportunityName: 'molestias',
                opportunityPartnerNames: [],
                opportunityType: OpportunityTypes.PRACTICUM,
                pathwayNames: ['Business Information Management', 'Diagnostic Services'],
                studentGradeLevel: '7',
                studentId: '10458576',
                studentName: 'Jon James',
                studentSisId: null,
              },
            ],
          },
        },
      },
    },
  },
};
