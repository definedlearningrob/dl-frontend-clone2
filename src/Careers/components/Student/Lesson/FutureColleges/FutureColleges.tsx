import { useTranslation } from 'react-i18next';

import SchoolBuilding from '@dc/images/school_building.svg';
import useUserInfo from '@dc/hooks/useUserInfo';
import { TStudentInfo } from '@dc/graphql/student/queries/userInfo';

import Card from '@shared/components/Card/Card';
import Image from '@shared/components/Image/Image';
import Link from '@shared/components/Link';

type Props = {
  careerName: string;
  hasInstitutionsInStudentState: boolean;
};

export const FutureColleges = ({ careerName, hasInstitutionsInStudentState }: Props) => {
  const { t } = useTranslation();
  const {
    userInfo: { state },
  } = useUserInfo<TStudentInfo>();
  const linkToInstitutionList = [
    `/post-secondary/search?keyword=${careerName}`,
    hasInstitutionsInStudentState && `location=${state}`,
  ]
    .filter(Boolean)
    .join('&');

  return (
    <Card className='bg-neutral-200 radius-md p-md gap-md'>
      <div className='xxxl:flex gap-md'>
        <div className='width-fit mb-base'>
          <Image data-testid='collages-and-future-image' src={SchoolBuilding} />
        </div>
        <div className='flex-1'>
          <h4>{t('course.collegesFuture.heading')}</h4>
          <p className='text-sm font-regular leading-lg mb-base'>
            {t('course.collegesFuture.text')}
          </p>
          <Link target='_blank' to={linkToInstitutionList} variant='primary-outlined'>
            {t('course.collegesFuture.button')}
          </Link>
        </div>
      </div>
    </Card>
  );
};
