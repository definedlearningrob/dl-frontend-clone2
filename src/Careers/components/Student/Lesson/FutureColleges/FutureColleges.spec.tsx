import { MockedProvider } from '@apollo/client/testing';
import { screen } from '@testing-library/dom';

import { FutureColleges } from '@dc/components/Student/Lesson/FutureColleges/FutureColleges';
import { renderWithRouterAndReduxProvider } from '@dc/utils/test';
import { UserInfoProvider } from '@dc/hooks/useUserInfo';
import { studentInfoMock } from '@dc/tests/mocks/studentMocks';

type Props = {
  hasInstitutionsInStudentState: boolean;
};

const renderLesson = ({ hasInstitutionsInStudentState }: Props) =>
  renderWithRouterAndReduxProvider(
    <MockedProvider mocks={[studentInfoMock]}>
      <UserInfoProvider value={{ userInfo: studentInfoMock.result.data.userInfo }}>
        <FutureColleges
          careerName="'Preschool Teachers, Except Special Education'"
          hasInstitutionsInStudentState={hasInstitutionsInStudentState}
        />
      </UserInfoProvider>
    </MockedProvider>,
    { initialState: { session: { user: { type: 'student' }, loginError: {} } } }
  );

describe('FutureColleges', () => {
  it('should render correctly', async () => {
    const { container } = renderLesson({ hasInstitutionsInStudentState: true });

    const heading = await screen.findByRole('heading', { name: 'Colleges & Future' });
    const collagesAndFutureDescription = await screen.findByText(
      'Find associated colleges, universities, trade programs, bootcamps and more with this course.'
    );

    const seeCollagesButton = await screen.findByRole('button', {
      name: 'See colleges & futures',
    });

    expect(heading).toBeInTheDocument();
    expect(collagesAndFutureDescription).toBeInTheDocument();
    expect(seeCollagesButton).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render when hasInstitutionsInStudentState is true', async () => {
    renderLesson({ hasInstitutionsInStudentState: true });

    const heading = screen.getByRole('heading', { name: 'Colleges & Future' });
    const collagesAndFutureDescription = screen.getByText(
      'Find associated colleges, universities, trade programs, bootcamps and more with this course.'
    );

    const seeCollagesLink = await screen.getByRole('link', {
      name: 'See colleges & futures',
    });

    expect(heading).toBeInTheDocument();
    expect(collagesAndFutureDescription).toBeInTheDocument();
    expect(seeCollagesLink).toHaveAttribute(
      'href',
      "/post-secondary/search?keyword='Preschool Teachers, Except Special Education'&location=ALASKA"
    );
  });

  it('should render when hasInstitutionsInStudentState is false', async () => {
    renderLesson({ hasInstitutionsInStudentState: false });

    const heading = await screen.findByRole('heading', { name: 'Colleges & Future' });
    const collagesAndFutureDescription = await screen.findByText(
      'Find associated colleges, universities, trade programs, bootcamps and more with this course.'
    );

    const seeCollagesLink = await screen.getByRole('link', {
      name: 'See colleges & futures',
    });

    expect(heading).toBeInTheDocument();
    expect(collagesAndFutureDescription).toBeInTheDocument();
    expect(seeCollagesLink).toHaveAttribute(
      'href',
      "/post-secondary/search?keyword='Preschool Teachers, Except Special Education'"
    );
  });
});
