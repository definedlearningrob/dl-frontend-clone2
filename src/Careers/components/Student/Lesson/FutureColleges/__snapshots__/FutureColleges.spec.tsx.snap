// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FutureColleges should render correctly 1`] = `
<div>
  <div
    class="card bg-neutral-200 radius-md p-md gap-md"
  >
    <div
      class="xxxl:flex gap-md"
    >
      <div
        class="width-fit mb-base"
      >
        <img
          data-testid="collages-and-future-image"
          src="SvgrURL"
        />
      </div>
      <div
        class="flex-1"
      >
        <h4>
          Colleges & Future
        </h4>
        <p
          class="text-sm font-regular leading-lg mb-base"
        >
          Find associated colleges, universities, trade programs, bootcamps and more with this course.
        </p>
        <a
          href="/post-secondary/search?keyword='Preschool Teachers, Except Special Education'&location=ALASKA"
          target="_blank"
        >
          <button
            class="button flex items-center justify-center gap-xs rounded-sm cursor-pointer text-sm font-medium no-underline focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] -primary-outlined text-primary-500 hover:bg-primary-200 py-xs px-base leading-[22px] border"
            data-testid="button"
            type="button"
          >
            <span
              class="items-center flex justify-center gap-xs"
            >
              See colleges & futures
            </span>
          </button>
        </a>
      </div>
    </div>
  </div>
</div>
`;
